src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -128,6 +128,11 @@
      */
     // TODO remove once baseline Git plugin 3.0.2+
     private static final AtomicLong jenkins41244Warning = new AtomicLong();
+    /**
+     * How long to delay events received from GitHub in order to allow the API caches to sync.
+     */
+    private static /*mostly final*/ int eventDelaySeconds =
+            Math.min(300, Math.max(0, Integer.getInteger(GitHubSCMSource.class.getName() + ".eventDelaySeconds", 5)));
```

Comment:
Isn't there a system class in core that @oleg-nenashev wrote that we are supposed to use now? Or maybe that is in a newer core than this plugin is written for.